sandbox.go

```diff
@@ -111,6 +111,17 @@ type containerConfig struct {
 	useDefaultSandBox bool
 	useExternalKey    bool
 	prio              int // higher the value, more the priority
+	dataDir           string
+}
+
+// newContainerConfig initializes the configuration
+// using the defaultPrefix as the data directory.
+// That option can be overriden by using OptionDataDir
+// when a new sandbox is created.
+func newContainerConfig() containerConfig {
+	return containerConfig{
+		dataDir: defaultPrefix,
```

Comment:
@calavera I don't think we need another dataDir config for every container. It's kind of daemon wide configuration and so it's appropriate to have it in `controller` and use it everywhere. In fact we already have this option as part of `controller`  here: https://github.com/docker/libnetwork/blob/master/config/config.go#L25. 

So we should make use of it and rework this PR based on that config?